openghg/dataobjects/_searchresults.py

```diff
@@ -95,290 +88,107 @@ def from_json(cls: Type[T], data: Union[bytes, str]) -> T:
         """
         loaded = json.loads(data)
 
-        return cls(results=loaded["results"], ranked_data=loaded["ranked_data"])
-
-    def rankings(self) -> Dict:
-        if not self.ranked_data:
-            print("No rank data")
-
-        rank_result = aDict()
-
-        for site, species_data in self.results.items():
-            for species, data in species_data.items():
-                rank_result[site][species] = data["rank_metadata"]
-
-        to_return: Dict = rank_result.to_dict()
-
-        return to_return
-
-    def raw(self) -> Dict:
-        """Returns the raw results data
-
-        Returns:
-            dict: Dictionary of results returned from search function
-        """
-        return self.results
-
-    def keys(self, site: str, species: str, inlet: Optional[str] = None) -> Optional[List[str]]:
-        """Return the data keys for the specified site and species.
-        This is intended mainly for use in the search function when filling
-        gaps of unranked dateranges.
-
-            Args:
-                site: Three letter site code
-                species: Species name
-                inlet: Inlet height, required for unranked data
-            Returns:
-                list: List of keys
-        """
-        site = site.lower()
-        species = species.lower()
-
-        if inlet is not None:
-            inlet = inlet.lower()
-
-        try:
-            if self.ranked_data:
-                keys: List = self.results[site][species]["keys"]
-            else:
-                keys = self.results[site][species][inlet]["keys"]
-
-            return keys
-        except KeyError:
-            print(f"No keys found for {species} at {site}")
-            return None
+        return cls(keys=loaded["keys"], metadata=loaded["metadata"])
 
-    def metadata(self, site: str, species: str, inlet: Optional[str] = None) -> Optional[Dict]:
-        """Return the metadata for the specified site and species
+    def retrieve(
+        self,
+        dataframe: Optional[DataFrame] = None,
+        sort: bool = True,
+        elevate_inlet: bool = False,
```

Comment:
Would be useful to have an explanation of additional inputs for the `retrieve()` function in the doc string - e.g. What does `elevate_inlet` mean?

Reply:
Done